hudi-common/src/main/java/org/apache/hudi/common/table/timeline/HoodieGlobalTimeline.java

```diff
@@ -0,0 +1,203 @@
+  public HoodieTimeline filterPendingLogCompactionTimeline() {
+    // override for efficiency
+    return this.activeTimeline.filterPendingLogCompactionTimeline();
+  }
+
+  @Override
+  public HoodieTimeline filterPendingMajorOrMinorCompactionTimeline() {
+    // override for efficiency
+    return this.activeTimeline.filterPendingMajorOrMinorCompactionTimeline();
+  }
+
+  @Override
+  public HoodieTimeline filterPendingReplaceTimeline() {
+    // override for efficiency
+    return this.activeTimeline.filterPendingReplaceTimeline();
+  }
+
+  @Override
+  public HoodieTimeline filterPendingRollbackTimeline() {
+    // override for efficiency
+    return this.activeTimeline.filterPendingRollbackTimeline();
+  }
+
+  @Override
+  public HoodieTimeline filterRequestedRollbackTimeline() {
+    // override for efficiency
+    return this.activeTimeline.filterRequestedRollbackTimeline();
+  }
+
+  @Override
+  public HoodieTimeline filterPendingIndexTimeline() {
+    // override for efficiency
+    return this.activeTimeline.filterPendingIndexTimeline();
+  }
+
+  @Override
+  public boolean empty() {
+    return this.activeTimeline.empty();
+  }
+
+  /**
+   * Returns whether the active timeline contains the given instant or the instant is archived.
+   * Needs to rethink the new semantics and rename this method with global timeline introduced.
+   */
+  @Override
+  public boolean containsOrBeforeTimelineStarts(String ts) {
+    return this.activeTimeline.containsOrBeforeTimelineStarts(ts);
+  }
+
+  @Override
+  public boolean isBeforeTimelineStarts(String ts) {
+    return this.activeTimeline.isBeforeTimelineStarts(ts);
+  }
+
+  @Override
+  public Option<HoodieInstant> getFirstNonSavepointCommit() {
+    return this.activeTimeline.getFirstNonSavepointCommit();
+  }
+
+  @Override
+  public Option<HoodieInstant> getLastPendingClusterInstant() {
+    // override for efficiency
+    return this.activeTimeline.getLastPendingClusterInstant();
+  }
+
+  /**
+   * Needs to fix this method to only check on active timeline.
+   */
+  @Override
+  protected Option<HoodieInstant> findFirstNonSavepointCommit(List<HoodieInstant> instants) {
+    return this.activeTimeline.findFirstNonSavepointCommit(instants);
+  }
+
+  /**
+   * Reloads all the instants.
+   */
+  public HoodieGlobalTimeline reload() {
+    return new HoodieGlobalTimeline(this.metaClient, Option.empty());
+  }
+
+  /**
+   * The active timeline is always reloaded,
+   * the archived timeline is reloaded based on the given start timestamp {@code startTs}.
+   */
+  public HoodieGlobalTimeline reload(String startTs) {
+    HoodieActiveTimeline reloadedActiveTimeline = this.metaClient.reloadActiveTimeline();
+    HoodieInstant oldFirstActiveCommit = this.activeTimeline.firstInstant().orElse(null);
+    HoodieInstant newFirstActiveCommit = reloadedActiveTimeline.firstInstant().orElse(null);
+    // reload the archived timeline incrementally if the archiving snapshot does not change.
```

Comment:
some way to only reload newer instants in the archived timeline? without reloading the whole tree again?

Reply:
Yeah, the archived timeline memo a cursor instant there, when we know the archiving is just the the same as before, we can do incremental loading for efficiency.